src/test/rustdoc/intra-doc-link-generic-params.rs

```diff
@@ -0,0 +1,10 @@
+#![crate_name = "foo"]
+
+//! Here's a link to [`Vec<T>`] and one to [`Box<Vec<Option<T>>>`].
+//!
+//! And what about a link to just [`Option`] and, with the generic, [`Option<T>`]?
+
+// @has foo/index.html '//a[@href="https://doc.rust-lang.org/nightly/alloc/vec/struct.Vec.html"]' 'Vec<T>'
+// @has foo/index.html '//a[@href="https://doc.rust-lang.org/nightly/alloc/boxed/struct.Box.html"]' 'Box<Vec<Option<T>>>'
+// @has foo/index.html '//a[@href="https://doc.rust-lang.org/nightly/core/option/enum.Option.html"]' 'Option'
+// @has foo/index.html '//a[@href="https://doc.rust-lang.org/nightly/core/option/enum.Option.html"]' 'Option<T>'
```

Comment:
The link text testing doesn't seem to be working; I deleted one of the `>` and it passed.

Reply:
That sounds like you're generating the wrong link text in rustdoc ;)

Reply:
It passes with **and** without the `>` though...

Reply:
I think it allows the text to be shorter, but not longer. So if the actual text was `Option<T` and you wrote `Option<T>`, it would fail the test. But since the actual was `Option<T>` and the expected was `Option<T`, it passed the test.

Maybe we should reconsider how that works? That seems like a separate PR though.

Reply:
That seems right; adding an *extra* `>` makes the test fail. I agree that we should reconsider that; it's kind of counter-intuitive.

Reply:
Opened https://github.com/rust-lang/rust/issues/76942